diff --git a/node_modules/@gitlab/ui/src/components/base/new_dropdowns/disclosure/disclosure_dropdown.vue b/node_modules/@gitlab/ui/src/components/base/new_dropdowns/disclosure/disclosure_dropdown.vue
index 277c89c..7c61044 100644
--- a/node_modules/@gitlab/ui/src/components/base/new_dropdowns/disclosure/disclosure_dropdown.vue
+++ b/node_modules/@gitlab/ui/src/components/base/new_dropdowns/disclosure/disclosure_dropdown.vue
@@ -347,7 +347,7 @@ export default {
       if (
         this.autoClose &&
         e.target.closest(ITEM_SELECTOR) &&
-        this.$refs.baseDropdown.containsElement(e.target)
+        this.$refs.baseDropdown.containsElement?.(e.target)
       ) {
         this.closeAndFocus();
       }
