diff --git a/node_modules/@vue/compat/dist/vue.cjs.js b/node_modules/@vue/compat/dist/vue.cjs.js
index aaaaaaa..bbbbbbb 100644
--- a/node_modules/@vue/compat/dist/vue.cjs.js
+++ b/node_modules/@vue/compat/dist/vue.cjs.js
@@ -9340,6 +9340,7 @@
     return normalizeSlotValue(rawSlot(...args));
   }, ctx);
   normalized._c = false;
+  if (rawSlot._ns) normalized._ns = true;
   return normalized;
 };
 const normalizeObjectSlots = (rawSlots, slots, instance) => {
diff --git a/node_modules/@vue/compat/dist/vue.esm-bundler.js b/node_modules/@vue/compat/dist/vue.esm-bundler.js
index aaaaaaa..bbbbbbb 100644
--- a/node_modules/@vue/compat/dist/vue.esm-bundler.js
+++ b/node_modules/@vue/compat/dist/vue.esm-bundler.js
@@ -9324,6 +9324,7 @@
     return normalizeSlotValue(rawSlot(...args));
   }, ctx);
   normalized._c = false;
+  if (rawSlot._ns) normalized._ns = true;
   return normalized;
 };
 const normalizeObjectSlots = (rawSlots, slots, instance) => {
diff --git a/node_modules/@vue/compat/dist/vue.runtime.esm-bundler.js b/node_modules/@vue/compat/dist/vue.runtime.esm-bundler.js
index aaaaaaa..bbbbbbb 100644
--- a/node_modules/@vue/compat/dist/vue.runtime.esm-bundler.js
+++ b/node_modules/@vue/compat/dist/vue.runtime.esm-bundler.js
@@ -9251,6 +9251,7 @@
     return normalizeSlotValue(rawSlot(...args));
   }, ctx);
   normalized._c = false;
+  if (rawSlot._ns) normalized._ns = true;
   return normalized;
 };
 const normalizeObjectSlots = (rawSlots, slots, instance) => {
