diff --git a/node_modules/@vue/compat/dist/vue.cjs.js b/node_modules/@vue/compat/dist/vue.cjs.js
index 00c2cda..1347235 100644
--- a/node_modules/@vue/compat/dist/vue.cjs.js
+++ b/node_modules/@vue/compat/dist/vue.cjs.js
@@ -6095,7 +6095,11 @@ function convertLegacyProps(legacyProps, type) {
         }
       }
     } else if (!skipLegacyRootLevelProps(key)) {
-      converted[key] = legacyProps[key];
+      if (isOn(key) && converted[key] && converted[key] !== legacyProps[key]) {
+        converted[key] = [].concat(converted[key], legacyProps[key]);
+      } else {
+        converted[key] = legacyProps[key];
+      }
     }
   }
   if (legacyProps.staticClass) {
diff --git a/node_modules/@vue/compat/dist/vue.runtime.esm-bundler.js b/node_modules/@vue/compat/dist/vue.runtime.esm-bundler.js
index f348d05..fe489df 100644
--- a/node_modules/@vue/compat/dist/vue.runtime.esm-bundler.js
+++ b/node_modules/@vue/compat/dist/vue.runtime.esm-bundler.js
@@ -6002,7 +6002,11 @@ function convertLegacyProps(legacyProps, type) {
         }
       }
     } else if (!skipLegacyRootLevelProps(key)) {
-      converted[key] = legacyProps[key];
+      if (isOn(key) && converted[key] && converted[key] !== legacyProps[key]) {
+        converted[key] = [].concat(converted[key], legacyProps[key]);
+      } else {
+        converted[key] = legacyProps[key];
+      }
     }
   }
   if (legacyProps.staticClass) {
