diff --git a/node_modules/@vue/compat/dist/vue.cjs.js b/node_modules/@vue/compat/dist/vue.cjs.js
index aaaaaaa..bbbbbbb 100644
--- a/node_modules/@vue/compat/dist/vue.cjs.js
+++ b/node_modules/@vue/compat/dist/vue.cjs.js
@@ -6134,7 +6134,14 @@
         }
       }
     } else if (!skipLegacyRootLevelProps(key)) {
-      converted[key] = legacyProps[key];
+      if (isOn(key) && converted[key] && converted[key] !== legacyProps[key]) {
+        converted[key] = [].concat(
+          converted[key],
+          legacyProps[key]
+        );
+      } else {
+        converted[key] = legacyProps[key];
+      }
     }
   }
   if (legacyProps.staticClass) {
diff --git a/node_modules/@vue/compat/dist/vue.esm-bundler.js b/node_modules/@vue/compat/dist/vue.esm-bundler.js
index aaaaaaa..bbbbbbb 100644
--- a/node_modules/@vue/compat/dist/vue.esm-bundler.js
+++ b/node_modules/@vue/compat/dist/vue.esm-bundler.js
@@ -6114,7 +6114,14 @@
         }
       }
     } else if (!skipLegacyRootLevelProps(key)) {
-      converted[key] = legacyProps[key];
+      if (isOn(key) && converted[key] && converted[key] !== legacyProps[key]) {
+        converted[key] = [].concat(
+          converted[key],
+          legacyProps[key]
+        );
+      } else {
+        converted[key] = legacyProps[key];
+      }
     }
   }
   if (legacyProps.staticClass) {
diff --git a/node_modules/@vue/compat/dist/vue.runtime.esm-bundler.js b/node_modules/@vue/compat/dist/vue.runtime.esm-bundler.js
index aaaaaaa..bbbbbbb 100644
--- a/node_modules/@vue/compat/dist/vue.runtime.esm-bundler.js
+++ b/node_modules/@vue/compat/dist/vue.runtime.esm-bundler.js
@@ -6041,7 +6041,14 @@
         }
       }
     } else if (!skipLegacyRootLevelProps(key)) {
-      converted[key] = legacyProps[key];
+      if (isOn(key) && converted[key] && converted[key] !== legacyProps[key]) {
+        converted[key] = [].concat(
+          converted[key],
+          legacyProps[key]
+        );
+      } else {
+        converted[key] = legacyProps[key];
+      }
     }
   }
   if (legacyProps.staticClass) {
