diff --git a/app/helpers/projects_helper.rb b/app/helpers/projects_helper.rb
index 04591476b72d..aecb81bb664b 100644
--- a/app/helpers/projects_helper.rb
+++ b/app/helpers/projects_helper.rb
@@ -404,7 +404,6 @@ def show_terraform_banner?(project)
   end
 
   def show_lfs_misconfiguration_banner?(project)
-    return false unless Feature.enabled?(:lfs_misconfiguration_banner)
     return false unless project.repository
     return false unless project.lfs_enabled?
 
diff --git a/spec/helpers/projects_helper_spec.rb b/spec/helpers/projects_helper_spec.rb
index a6ea3ec16179..ea2fdf3f4ba7 100644
--- a/spec/helpers/projects_helper_spec.rb
+++ b/spec/helpers/projects_helper_spec.rb
@@ -903,14 +903,6 @@ def license_name
 
           it { is_expected.to be_falsey }
         end
-
-        context 'when lfs_misconfiguration_banner feature flag is disabled' do
-          before do
-            stub_feature_flags(lfs_misconfiguration_banner: false)
-          end
-
-          it { is_expected.to be_falsey }
-        end
       end
 
       context 'when it does have a .gitattributes file' do
